collections/_drafts/agent-forwarding.md

```diff
@@ -0,0 +1,275 @@
+other GnuPG modules how to connect to the agent. However, correctly managing the
+startup and this environment variable is cumbersome, so a more straightforward
+method is required. Since GnuPG 2.0.16, the \--use-standard-socket option
+already allowed starting the agent on the fly; however, the environment variable
+was still needed.
+
+With GnuPG 2.1, the need for GPG_AGENT_INFO has been completely removed, and the
+variable is ignored.
+
+Instead, a fixed Unix domain socket named S.gpg\-agent in the GnuPG home
+directory (by default ~/.gnupg) is used. The agent is also started on-demand by
+all tools requiring services from the agent.
+
+If the option \--enable-ssh\-support is used, the auto-start mechanism does not
+work because _ssh_ does not know about this mechanism. Instead, the environment
+variable SSH_AUTH_SOCK must be set to the S.gpg-agent.ssh socket in the GnuPG
+home directory. Further, _gpg\_\_\-agent_ must be started by either using a
+GnuPG command that implicitly starts gpg\-agent or by using \`gpgconf --launch
+gpg\-agent\` to explicitly start it without first having to use a GnuPG command.
+
+gpg\-agent is a daemon to manage secret (private) keys independently from any
+protocol. It is a backend for gpg, gpgsm, and other utilities.
+
+GPG Agent Configuration
+
+There are a few configuration files needed for the operation of the agent. They
+may all be found in the current GnuPG home directory, which defaults
+to ~/.gnupg.
+
+\[\[windows\]\]
+
+However, there may be problems on Windows systems with Gpg4Win installed;
+**Gpg4Win may have changed this default GnuPG home directory location** to an
+AppData subdirectory (i.e., C:\\Users\\<username>\\AppData\\Roaming\\gnupg).
+This location, however, is not the location our GPG agent will be using, so to
+reaffirm our preference for the default location, we will set
+the GNUPGHOME environment variable to ~/.gnupg in the Git Bash startup script by
+running the following.
+
+echo 'export GNUPGHOME="~/.gnupg"' >> .bashrc
+
+\[\[endwindows\]\]
+
+Setting environment variables
+
+Add the following lines to your .bashrc or whatever initialization file is used
+for all shell invocations:
+
+GPG_TTY=$(tty)
+
+export GPG_TTY
+
+This variable may only be helpful if you use \`pinentry\-curses\` (the
+terminal-based pin entry program).
+
+\[\[windows\]\]
+
+On Windows systems, you should add the above lines to the ~/.bashrc file for use
+by Git Bash.
+
+\[\[endwindows\]\]
+
+GnuPG configuration
+
+It's important to note that GnuPG on the remote system still needs your public
+GPG keys to work correctly. So you have to ensure they are available on the
+remote system even if your secret keys are not.
+
+\[\[note\]\]
+
+If you use VSCode with the remote extension pack, you may skip this step and
+move on to the next section. This step of copying over your local public GPG
+keyring into the remote container gets done automatically by the extension.
+
+\[\[endnote\]\]
```

Comment:
```suggestion
<aside class="note">

<b>Note:</b>If you use VSCode with the remote extension pack, you may skip this step and
move on to the next section. This step of copying over your local public GPG
keyring into the remote container gets done automatically by the extension.

</aside>
```

Reply:
that attempt to get this done w/o using `picture` element
was simply us trying on Google's not-very-good way of going about it

Reply:
> [&hellip;] should've been a directive.
>
> Square brackets are used a lot already in markdown [&hellip;]
> &mdash;https://twitter.com/wooorm/status/1701865524236345734

_Originally posted by @DerekNonGeneric in https://github.com/OpenINF/openinf.github.io/pull/1106#discussion_r1470135406_

🔗 https://talk.commonmark.org/t/generic-directives-plugins-syntax/444

that is wayyy too much reading, but skipping all of it would be fine, here is the one:

🔗 https://github.com/remarkjs/remark-directive?tab=readme-ov-file#example-styled-blocks

Reply:
```suggestion
:::note{.note}

If you use VSCode with the remote extension pack, you may skip this step and
move on to the next section. This step of copying over your local public GPG
keyring into the remote container gets done automatically by the extension.

:::
```